vertx-template-engines/vertx-web-templ-freemarker/src/main/java/io/vertx/ext/web/templ/impl/FreeMarkerTemplateEngineImpl.java

```diff
@@ -62,8 +63,9 @@ public FreeMarkerTemplateEngine setMaxCacheSize(int maxCacheSize) {
   }
 
   @Override
-  public void render(RoutingContext context, String templateFileName, Handler<AsyncResult<Buffer>> handler) {
+  public void render(RoutingContext context, String templateDirectory, String templateFileName, Handler<AsyncResult<Buffer>> handler) {
     try {
+      templateFileName = templateDirectory + templateFileName;
```

Comment:
I think that most users will not be calling this directly but rather thru `TemplateHandlerImpl`, so this should be left to caller - there is similarly no check if `templateFileName` is `null` or empty ... 
As I understand both you and @yunyu also came to the same conclusion ..?